app/lib/shared/analyzer_client.dart

```diff
@@ -65,4 +86,43 @@ class AnalysisView {
     list.sort();
     return list;
   }
+
+  /// Returns the raw health score between -1.0 and 1.0
+  /// These need to be normalized for search and frontend purposes.
+  double get health {
+    if (_data == null) return 0.0; // missing analysis
+    if (!hasAnalysisData) return -1.0; // aborted analysis
+
+    // Starting out with max.
+    double score = 1.0;
+
+    // Penalty for each major issue.
+    if (_summary.issues != null && _summary.issues.isNotEmpty) {
+      score -= _summary.issues.length * 0.2;
+    }
+
+    if (_summary.dartFiles != null) {
+      for (DartFileSummary dfs in _summary.dartFiles.values) {
+        // penalties only for files inside the library
+        if (!dfs.isInLib) continue;
+
+        // penalty for formatting
+        if (!dfs.isFormatted != true) {
+          score -= 0.001;
+        }
```

Comment:
That is true, and I have no idea yet how we want to address that. Both filesize-based normalization and ignoring it altogether has its pros and cons. Luckily this is easy to change later if we make up our mind to do a different thing.